### PR TITLE
Initupdate

### PR DESCRIPTION
**Pull request by cathyyfyang**
# Related Issue

-   issue goes here

# Proposed Changes

-   change the port

# Additional Info

-   add pr template

# Checklist

-   [ ] eslint+prettier+husky
-   [ ] styled-component
-   [ ] directories classification
-   [ ] redux
-   [ ] cross-env PORT=8000

# Screenshots

|         Original          |         Updated         |
| :-----------------------: | :---------------------: |
| ** orginial screenshot ** | ** updatd screenshot ** |
Screen Shot 2022-09-18 at 6 07 44 pm
Screen Shot 2022-09-18 at 6 06 59 pm

### CODE REVIEW

**Comment by Billy-MC** [comment]
> # Related Issue
> * issue goes here
> 
> # Proposed Changes
> * change the port
> 
> # Additional Info
> * add pr template
> 
> # Checklist
> * [ ]  eslint+prettier+husky
> * [ ]  styled-component
> * [ ]  directories classification
> * [ ]  redux
> * [ ]  cross-env PORT=8000
> 
> # Screenshots
> Original	Updated
> ** orginial screenshot **	** updatd screenshot **
> Screen Shot 2022-09-18 at 6 07 44 pm
> 
> Screen Shot 2022-09-18 at 6 06 59 pm

PR template 要有Description, 每次PR 都需要描述這個PR做了什麼

**Comment by Billy-MC** [comment]
> # Related Issue
> * issue goes here
> 
> # Proposed Changes
> * change the port
> 
> # Additional Info
> * add pr template
> 
> # Checklist
> * [ ]  eslint+prettier+husky
> * [ ]  styled-component
> * [ ]  directories classification
> * [ ]  redux
> * [ ]  cross-env PORT=8000
> 
> # Screenshots
> Original	Updated
> ** orginial screenshot **	** updatd screenshot **
> Screen Shot 2022-09-18 at 6 07 44 pm
> 
> Screen Shot 2022-09-18 at 6 06 59 pm

checklist錯了, 不是你安裝了什麼, 是你PR時, 有做什麼, 例如unit-test, e2e-test 之類, 你可以再看看別的template如何寫

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+	"parserOptions": {
+		"ecmaFeatures": {
+			"jsx": true
+		},
+		"ecmaVersion": "latest",
+		"sourceType": "module"
+	},
+	"plugins": ["react", "jest"],
+	"rules": {
+		"no-underscore-dangle": 0,
```
Comment (Billy-MC) [inline .eslintrc.json:23]: no underscore 不應該off的, 應該需要放一些allow進去

File: .eslintrc.json [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+	"env": {
+		"browser": true,
+		"es2021": true,
+		"jest/globals": true
+	},
+	"extends": [
+		"plugin:react/recommended",
+		"airbnb",
+		"prettier",
+		"plugin:jest/recommended"
```
Comment (Billy-MC) [inline .eslintrc.json:11]: jsx-a11y的extends和plugins都應該需要有

File: .eslintrc.json [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+	],
+	"overrides": [],
+	"parserOptions": {
+		"ecmaFeatures": {
+			"jsx": true
+		},
+		"ecmaVersion": "latest",
+		"sourceType": "module"
+	},
+	"plugins": ["react", "jest"],
```
Comment (Billy-MC) [inline .eslintrc.json:21]: plugins少了jsx-a11y 和 prettier

File: .eslintrc.json [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+	"rules": {
+		"no-underscore-dangle": 0,
+		"import/extensions": ["error", "ignorePackages"],
+		"semi": ["error", "always"],
+		"linebreak-style": ["error", "unix"],
+		"quotes": ["error", "single"],
+		"no-tabs": 0,
+		"no-console": 1,
+		"no-shadow": "off",
```
Comment (Billy-MC) [inline .eslintrc.json:30]: no-shadow可以先error, 需要ignore 時再寫進去

File: .eslintrc.json [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+			{
+				"functions": true,
+				"classes": true,
+				"variables": true,
+				"allowNamedExports": false
+			}
+		],
+		"no-param-reassign": ["error", { "props": false }],
+		"sort-keys": "error",
+		"no-unused-vars": "error",
```
Comment (Billy-MC) [inline .eslintrc.json:42]: 如果你有force lint, no-unused-vars可以warn, 不然你comment out 一個東西時, import的就會eslint報錯

File: .github/pull_request_template.md [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+# Additional Info
+
+-   any additional information or context
+
+# Checklist
+
+-   [ ] Tests
+-   [ ] Translations
+-   [ ] Documentation
+
```
Comment (Billy-MC) [inline .github/pull_request_template.md:19]: 這個Checklist可以, 你每次PR 不用改的,
最好多一個Description, 去描述你當前的ticket在做什麼

File: .gitignore [hunk 7]
```diff
@@ -0,0 +1,23 @@
+# See https://help.github.com/articles/ignoring-files/ for more about ignoring files.
+
```
Comment (Billy-MC) [inline .gitignore:2]: 可以加.env

File: .prettierrc.json [hunk 8]
```diff
@@ -0,0 +1,9 @@
+{
+	"tabWidth": 4,
+	"useTabs": true,
+	"printWidth": 80,
```
Comment (Billy-MC) [inline .prettierrc.json:4]: 80太短了, 100~120會好一點

File: README.md [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+This project was bootstrapped with [Create React App](https://github.com/facebook/create-react-app).
+
+## Available Scripts
+
+In the project directory, you can run:
+
+### `npm start`
+
+Runs the app in the development mode.\
+Open [http://localhost:3000](http://localhost:3000) to view it in your browser.
```
Comment (Billy-MC) [inline README.md:12]: backend又3000, frontend又3000 會有衝突, 其中一邊必須要改別的

File: src/App/App.css [hunk 10]
```diff
@@ -0,0 +1,38 @@
+.App {
```
Comment (Billy-MC) [inline src/App/App.css:1]: 如果用styled-components, css file可以delete

File: src/App/App.jsx [hunk 11]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import logo from './logo.svg';
+import './App.css';
+
+function App() {
+	return (
+		<div className="App">
```
Comment (Billy-MC) [inline src/App/App.jsx:7]: 用styled-components 可以把所有className delete

**Comment by Billy-MC** [comment]
@cathyyfyang  
1. components文件夾中的3個 Folder代表什麼? components? 還是pages? 如果是Pages 就放錯了
2. 另外 react-router-dom和redux-toolket沒有安裝
3. 不需要有一個叫App 的folder 
4. 一個前端的文件架構 需要有 router, pages/screens, utils, layouts, hooks, assets, services, store(如果你們用redux

**Comment by Billy-MC** [comment]
@cathyyfyang 另外mui+styled-components 是需要配置的，而還沒有配好的
可以看一下MUI文檔： https://mui.com/material-ui/guides/styled-engine/

File: package.json [hunk 12]
```diff
@@ -0,0 +1,79 @@
...
+		"react": "^18.2.0",
+		"react-dom": "^18.2.0",
+		"react-redux": "^8.0.2",
+		"react-scripts": "5.0.1",
+		"redux": "^4.2.0",
+		"styled-components": "^5.3.5",
+		"web-vitals": "^2.1.4"
+	},
+	"scripts": {
+		"start": "cross-env PORT=8000 react-scripts start",
```
Comment (Billy-MC) [inline package.json:20]: 嗯看到這裡有改port了, 但README也要改, 要學寫一下documentation

File: .eslintrc.json (hunk 1 above)
Comment (cathyyfyang) [reply .eslintrc.json:23]: done

File: .eslintrc.json (hunk 2 above)
Comment (cathyyfyang) [reply .eslintrc.json:11]: Done

File: .eslintrc.json (hunk 3 above)
Comment (cathyyfyang) [reply .eslintrc.json:21]: Done

File: .eslintrc.json (hunk 5 above)
Comment (cathyyfyang) [reply .eslintrc.json:42]: 我有lint -- --fix，这个warn怎么弄，要加什么条件呢？我加了不需要检查的异常 no-unused-vars: ["error", { "varsIgnorePattern": "[iI]gnored" }]，看看可不可以了。

File: .github/pull_request_template.md (hunk 6 above)
Comment (cathyyfyang) [reply .github/pull_request_template.md:19]: Done

File: .gitignore (hunk 7 above)
Comment (cathyyfyang) [reply .gitignore:2]: Done

File: .prettierrc.json (hunk 8 above)
Comment (cathyyfyang) [reply .prettierrc.json:4]: Done

File: README.md (hunk 9 above)
Comment (cathyyfyang) [reply README.md:12]: Done

File: src/App/App.css (hunk 10 above)
Comment (cathyyfyang) [reply src/App/App.css:1]: done

File: package.json (hunk 12 above)
Comment (cathyyfyang) [reply package.json:20]: Done

File: .eslintrc.json (hunk 4 above)
Comment (cathyyfyang) [reply .eslintrc.json:30]: Done

**Comment by Billy-MC** [comment]
我先decline, 你用KS-35那個branch重新提交pr, 記得寫好pr description

File: .eslintrc.json (hunk 5 above)
Comment (cathyyfyang) [reply .eslintrc.json:42]: done

File: src/App/App.jsx (hunk 11 above)
Comment (cathyyfyang) [reply src/App/App.jsx:7]: Done

**Pull request closed without merging**